### DISCUSSION TITLE
Better Fit for DRT Analysis

### DISCUSSION THREAD

**Discussion by DDDDMX** [vyrjana/DearEIS#22] [Q&A] [answered]
Hello,

I am quite new to DRT analysis and wanted to know if I could change the fit of the DRT Analysis. As you can see from the picture below, the fit from the automatic DRT analysis was not very optimal in the high-frequency range. I read that DRT is data sensitive and this unoptimized fitting would cause inaccuracy to the DRT plot, right?

My questions are:
- Is the inaccuracy of the DRT plot only on the high-frequency range or for the entire DRT plot?
- Can I optimize the fit (for example, connect the fit with the equivalent circuit for the DRT Analysis)?

Feel free to ask me anything if the information that I provided is insufficient :)

**Comment by vyrjana** [comment, marked as answer]
Which method (TR-NNLS, TR-RBF, etc.) and what settings are you using?

For example, the Tikhonov-regularized methods (TR-*) have a regularization parameter (lambda) that you might need to adjust manually. If the regularization parameter is very small, then the resulting impedance spectrum should closely resemble the data, but then the DRT may contain a lot of artifacts (e.g., many sharp peaks that shouldn't actually be there). If the regularization parameter is large, then the ability to resolve peaks in the DRT will be poor and will result in broad peaks. If there are multiple peaks close together, then they might meld together. So, there is a degree of compromise when using those methods.

There are several things about the experimental impedance spectrum itself that can cause issues for DRT analysis. [Boukamp](https://doi.org/10.1088/2515-7655/aba9e0) provides some advice such as replacing rather than simply omitting outliers. However, I think in your case the behavior of the imaginary part of the impedance at the frequency extremes is the problem.

Are you able to reliably measure the impedance at higher frequencies? Going to higher frequencies would most likely bring the imaginary part of the impedance closer to zero at that end of the experimental impedance spectrum, which should improve the quality of the DRT and therefore also the ability to recreate that part of the impedance spectrum based on the obtained DRT. However, you might run into issues with the potentiostat/galvanostat depending on a) its supported frequency range and b) the accuracy when trying to measure low impedances at high frequencies (check out [accuracy contour plots](https://www.gamry.com/application-notes/EIS/accuracy-contour-plots-measurement-and-discussion/) if you aren't familiar with them). The electrodes themselves may also complicate matters depending on your experimental setup. Cable inductance may also become a significant contributor to the total impedance when going to higher frequencies with low-impedance samples.

The imaginary part of the impedance is diverging from zero at the low-frequency end, which is not something that most methods can handle well. Some groups have preprocessed the impedance spectrum to obtain something that is easier to handle (e.g., fitting a fairly simple equivalent circuit to the low-frequency portion of the impedance spectrum and then subtracting the fitted model from the impedance spectrum). [Py et al.](https://doi.org/10.1016/j.electacta.2023.143741) described an approach for estimating the distribution of capacitive times (DCT) based on the admittance spectrum (Y = 1/Z). They were able to then handle impedance spectra where the imaginary impedance diverged from zero at low frequencies

**Reply by DDDDMX** [reply]
I used the TR-RBF method and attempted to vary the lambda parameter, but the fit did not improve. To investigate further, I examined the accuracy contour plots of the device, which suggest that the issue might stem from the measurement equipment itself. I set the frequency range to the maximum (200 kHz), and at this frequency, with the impedance value falling below 1 Ohm, the measurement accuracy already exceeds the limits defined by the contour plots.

When the imaginary part of the impedance diverges from zero at the low-frequency end, does this mean that the DRT analysis at this frequency is also unreliable? Will it always result in a high-intensity peak? Initially, I suspected had a Li+ diffusion problem because all of my cells have high-intensity peaks in the mHz range. However, after looking at the paper by Py et al., it seems this might not be the case. But I have also read some other works like [Malik et al.](https://iopscience.iop.org/article/10.1149/1945-7111/ad8d51) that analyze their cells directly from the DRT plot even if the impedance diverges from zero at the low-frequency end and concluded to the diffusivity problems, which confuses me. Thank you very much for the helpful answer. This has helped me a lot.

**Reply by vyrjana** [reply]
For example, if the divergence looked purely capacitive (straight, vertical line), then you could model it as just a capacitor. However, it could also be modeled as a parallel RC element with a very large R. One would need data points at even lower frequencies in order to figure out the "correct" model, but that may not be feasible for practical reasons such as drift when measuring for long periods of time.

Similarly, there are multiple models for describing diffusion impedance (e.g., different types of Warburg diffusion impedance, but there are also other models). For some models, the imaginary impedance approaches zero provided that one is able to measure at the low frequencies that are required to see this behavior. A ~45° degree line in the impedance Nyquist plot may result in a DRT that contains multiple peaks and their magnitudes increase as the time constant increases. If you look at the equations that describe the impedance based on a DRT, then you can see that the integral resembles the equation for the impedance of a parallel RC element. So, the imaginary part is assumed to approach zero at both frequency extremes.
